src/components/confirm/confirm.pw.tsx

```diff
@@ -0,0 +1,688 @@
+
+      const button = page
+        .getByRole("button")
+        .filter({ hasText: confirmButtonText });
+      await expect(button).toHaveAttribute("data-element", "confirm");
+    });
+  });
+
+  testData.forEach((cancelButtonText) => {
+    test(`should check cancel button text is ${cancelButtonText}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent cancelLabel={cancelButtonText} />);
+
+      const button = page
+        .getByRole("button")
+        .filter({ hasText: cancelButtonText });
+      await expect(button).toHaveAttribute("data-element", "cancel");
+    });
+  });
+
+  testData.forEach((titleText) => {
+    test(`should check title text is ${titleText}`, async ({ mount, page }) => {
+      await mount(<ConfirmComponent title={titleText} />);
+
+      await expect(page.getByText(titleText)).toHaveAttribute(
+        "data-element",
+        "title"
+      );
+    });
+  });
+
+  testData.forEach((subTitleText) => {
+    test(`should check subtitle text is ${subTitleText}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent subtitle={subTitleText} />);
+
+      await expect(page.getByText(subTitleText)).toHaveAttribute(
+        "data-element",
+        "subtitle"
+      );
+    });
+  });
+
+  heights.forEach(([heightnumber, heightstring]) => {
+    test(`should check Confirm height is ${heightstring}px`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent height={heightstring} />);
+
+      const viewportHeight = 768;
+
+      let resultHeight: number;
+      if (heightnumber >= viewportHeight - 20) {
+        resultHeight = viewportHeight - 20;
+      } else {
+        resultHeight = heightnumber;
+      }
+
+      await assertCssValueIsApproximately(
+        page.getByRole("alertdialog"),
+        "height",
+        resultHeight
+      );
+    });
+  });
+
+  ([
+    [SIZE.EXTRASMALL, 300],
+    [SIZE.SMALL, 380],
+    [SIZE.MEDIUMSMALL, 540],
+    [SIZE.MEDIUM, 750],
+    [SIZE.MEDIUMLARGE, 850],
+    [SIZE.LARGE, 960],
+    [SIZE.EXTRALARGE, 1080],
+  ] as [ConfirmProps["size"], number][]).forEach(([sizeName, size]) => {
+    test(`should check confirm size is %{sizeName} with width of ${size}px`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent size={sizeName} />);
+
+      await assertCssValueIsApproximately(
+        page.getByRole("alertdialog"),
+        "width",
+        size
+      );
+    });
+  });
+
+  ([
+    ["error", "error"],
+    ["warning", "warning"],
+  ] as [ConfirmProps["iconType"], string][]).forEach(([iconType, element]) => {
+    test(`should check confirm has ${iconType} icon`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent iconType={iconType} />);
+
+      const firstIcon = icon(page).first();
+      await expect(firstIcon).toHaveAttribute("data-element", element);
+    });
+  });
```

Comment:
Suggestion:

 (["error", "warning"]as const).forEach((type) => {
    test(`should check confirm has ${type} icon`, async ({
      mount,
      page,
    }) => {
      await mount(<ConfirmComponent iconType={type} />);

      const firstIcon = icon(page).first();
      await expect(firstIcon).toHaveAttribute("data-element", type);
    });
  });